### PR TITLE
Add kagglesdk

### PR DESCRIPTION
**Pull request by stevemessick**
This is generated code. To the extent that it has been tested, it is used in some of the public API code, which is exercised by the unit tests.

It might be nice if there was a way to allow both this repo and `kagglehub` to share this package without publishing to pypi.

### CODE REVIEW

**Review by jplotts — APPROVED** [review]
Nice work!  This will really help keep the API in sync with MT code.

File: kagglesdk/kaggle_env.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+from enum import Enum
+
+
+class KaggleEnv(Enum):
+  LOCAL = 0  # localhost
+  STAGING = 1  # staging.kaggle.com
+  ADMIN = 2  # admin.kaggle.com
+  QA = 3 # qa.kaggle.com
+  # Direct prod access is not allowed to have IAP protection during testing.
+  # PROD = 3  # www.kaggle.com
```
[the remark this answers is not in the document]
Comment (stevemessick) [reply kagglesdk/kaggle_env.py:11]: @jplotts These constants are used in several places. How about I remove the comment?

File: kagglesdk/kaggle_object.py [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    ObjectSerializer.__init__(self, lambda cls, v, _: v, lambda cls, v: v)
+
+
+# Adapted from https://stackoverflow.com/questions/1175208/elegant-python-function-to-convert-camelcase-to-snake-case
+_pascal_to_upper_snake_case_regex = re.compile('((?<=[a-z0-9])[A-Z]|(?!^)[A-Z](?=[a-z]))')
+
+
+def _pascal_case_to_upper_snake_case(string):
+  return _pascal_to_upper_snake_case_regex.sub(r'_\1', string).upper()
+
```
[the remark this answers is not in the document]
Comment (stevemessick) [reply kagglesdk/kaggle_object.py:26]: That's been in the code for two years.

File: kagglesdk/kaggle_object.py [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+                              lambda cls, v: EnumSerializer._from_str(cls, v))
+
+  @staticmethod
+  def _to_str(cls, v):
+    # "v" corresponds to an enum instance: Example foo or Foo.Test above.
+    # "cls" corresponds to the enum type Foo above.
+    #enum_prefix = f'{_pascal_case_to_upper_snake_case(cls.__name__)}_'
+    #if v.name.startswith(enum_prefix):
+    #  return v.name
+    #return f'{enum_prefix}{v.name}'
```
[the remark this answers is not in the document]
Comment (stevemessick) [reply kagglesdk/kaggle_object.py:50]: I don't know. They've been there a while.

**Comment by rosbo** [comment]
> It might be nice if there was a way to allow both this repo and kagglehub to share this package without publishing to pypi.

Should we publish a `kagglesdk` package to pypi?

Alternatively, if we want to make it less visible, we could publish `kagglesdk` only to a github repo and refer to it via a git URL?

Thoughts?

**Comment by stevemessick** [comment]
Putting it in pypi would be the easy way, but someone suggested we might not want to make it that visible. I like the idea of using github. When I (finally) revise the package management I'll look into that.

**Pull request merged**